packages/widget/src/controllers/transfers/fungible-token-transfer.ts

```diff
@@ -245,13 +253,49 @@ export class FungibleTokenTransferController implements ReactiveController {
             !this.supportedResources.includes(route.resource))
       )
       .map((route) => route.resource);
-    //unselect destination if equal to source network or isn't in list of available destination networks
-    if (
-      this.destinationNetwork?.id === sourceNetwork.id ||
-      !this.supportedDestinationNetworks.includes(this.destinationNetwork!)
-    ) {
+
+    // unselect destination if equal to source network or isn't in list of available destination networks
+    if (this.destinationNetwork?.id === sourceNetwork.id) {
       this.destinationNetwork = undefined;
+      this.selectedResource = undefined;
+    }
+
+    // imposible route source -> source
+    const isSourceOnSuportedDestinations =
+      this.supportedDestinationNetworks.some(
+        (destinationDomain) =>
+          destinationDomain.chainId === this.sourceNetwork?.chainId
+      );
+
+    if (isSourceOnSuportedDestinations) {
+      this.supportedDestinationNetworks = this.routesCache
+        .get(sourceNetwork.chainId)!
+        .filter(
+          (route) =>
+            route.toDomain.chainId !== sourceNetwork.chainId &&
+            !this.supportedDestinationNetworks.includes(route.toDomain)
+        )
+        .map((route) => route.toDomain);
     }
+
+    const hasSelectedResourceChange =
```

Comment:
Don't think you need this anymore, it will get automatically handled by amountSelector component now